Neos.Flow/Classes/Mvc/Routing/Dto/ActionUriSpecification.php

```diff
@@ -0,0 +1,169 @@
+<?php
+declare(strict_types=1);
+namespace Neos\Flow\Mvc\Routing\Dto;
+
+/*
+ * This file is part of the Neos.Flow package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+use Neos\Flow\Annotations as Flow;
+use Neos\Flow\Mvc\ActionRequest;
+use Neos\Utility\Arrays;
+use Psr\Http\Message\UriInterface;
+
+/**
+ * @Flow\Proxy(false)
+ */
+final class ActionUriSpecification
+{
+    private function __construct(
+        private string $actionName,
+        private string $controllerName,
+        private string $packageKey,
+        /** @deprecated with Flow 8.3 - The use of subpackage keys is discouraged and only supported for backwards compatibility */
+        private ?string $subpackageKey,
```

Comment:
@fcool and me wondered what the alternative of subpackageKey would look like ...